datafusion/proto/src/logical_plan/from_proto.rs

```diff
@@ -412,16 +411,6 @@ impl From<&protobuf::StringifiedPlan> for StringifiedPlan {
     }
 }
 
-impl From<&protobuf::ScalarFunction> for BuiltinScalarFunction {
```

Comment:
I think this is the correct change -- now all functions are resolved by name.

While this is a breaking API change, most other changes to the protobuf definitions are breaking changes -- see https://github.com/apache/datafusion/blob/70db5eab8996af4816958f798f6ee887dffb69ed/datafusion/proto/src/lib.rs#L37-L42